### PR TITLE
Exclude 'exclude-upstream-addresses' config parameter.

### PR DESCRIPTION
**Pull request by vlamug**
The following changes were made:
 - The 'exclude-upstream-addresses' config parameter was added to exclude upstream addresses.
 - The gathering of peer id metric was removed.

### CODE REVIEW

**Comment by vlamug** [comment]
@d-ulyanov , hello. Could you check it?
Thanks.

File: scraper/nginx_plus.go [hunk 1]
```diff
@@ -120,14 +121,25 @@ func (scr *NginxPlusScraper) scrapeUpstream(status *Status, metrics chan<- metri
 		}
 
 		for _, peer := range upstream.Peers {
+			if len(scr.excludeUpstreamAddresses) != 0 {
+				exclude := false
+				for _, address := range scr.excludeUpstreamAddresses {
```
Comment (d-ulyanov) [inline scraper/nginx_plus.go]: It seems that map[addr]bool suits better here

File: common/config.go [hunk 2]
```diff
@@ -2,20 +2,29 @@ package common
...
-	MetricsPath   string
-	Namespace     string
-	NginxUrls     []string
-	NginxPlusUrls []string
+	ListenAddress            string
+	MetricsPath              string
+	Namespace                string
+	NginxUrls                []string
+	NginxPlusUrls            []string
+	ExcludeUpstreamAddresses []string
```
Comment (d-ulyanov) [inline common/config.go]: May be excludeUpstreamPeers?

File: scraper/nginx_plus.go [hunk 3]
```diff
@@ -240,12 +252,25 @@ func (scr *NginxPlusScraper) scrapeStream(status *Status, metrics chan<- metric.
 		metrics <- metric.NewMetric("stream_upstream_zombies", upstream.Zombies, upstreamLabels)
 
 		for _, peer := range upstream.Peers {
+			if len(scr.excludeUpstreamAddresses) != 0 {
```
Comment (d-ulyanov) [inline scraper/nginx_plus.go]: Excess condition

File: main.go [hunk 4]
```diff
@@ -84,7 +86,7 @@ func registerExporter(namespace string, nginxUrls []string, nginxPlusUrls []stri
 	prometheus.MustRegister(exporter.NewNginxPlusExporter(
 		client,
 		scraper.NewNginxScraper(),
-		scraper.NewNginxPlusScraper(),
+		scraper.NewNginxPlusScraper(excludeUpstreamAddresses),
```
Comment (d-ulyanov) [inline main.go]: Let’s find another approach for initialization of this structure. 
Check jaeger tracing for good examples

**Review by d-ulyanov — CHANGES_REQUESTED** [review]

File: common/config.go (hunk 2 above)
Comment (vlamug) [reply common/config.go]: Ok.

File: scraper/nginx_plus.go (hunk 1 above)
Comment (vlamug) [reply scraper/nginx_plus.go]: Good idea.

File: scraper/nginx_plus.go (hunk 3 above)
Comment (vlamug) [reply scraper/nginx_plus.go]: No, it excludes stream addresses, but the above condition exclude upstream addresses.

**Pull request closed without merging**